### ISSUE TITLE
Generic Talon files and their Python files should live side-by-side

### ISSUE THREAD

**Issue by wenkokke** [knausj85/knausj_talon#669]
I would like for knausj_talon to move towards the following organizational principle:

> If a Talon file uses abstract actions, there should be a Python file with the same name in the same directory which declares them and describes their intended behavior in a comment.

I've applied this principle to the programming language modes in #652 and #656, and I think it makes the code a lot more readable and discoverable for people who are not already intimately familiar with the code base.

I am opening this issue as a place to have a discussion about this.

**Comment by rntz** [comment]
I generally agree. I'm not sure how rigidly to apply this principle, but I definitely feel like the weird splitting of some functionality between `misc/` and `code/` (eg. `misc/formatters.talon` vs `code/formatters.py`) is a mistake, and it would be easier to understand/navigate if they were in the same directory.

**Comment by auscompgeek** [comment]
I'd like to go a bit further and suggest that the text related files should all go in `text/`. `formatters` is a good example.

**Comment by wenkokke** [comment]
Would there be interest in small piecemeal pull requests to reorganize the repository along the lines that I have organized my own repository? See: <https://github.com/wenkokke/talon-user>

```sh
apps/           # contains Talon scripts associated with applications, e.g., Firefox;
core/           # contains Talon scripts which are always running;
tags/           # contains Talon scripts which are activated by a certain tag;
tags/code/      # contains Talon scripts which are associated with a programming language, e.g., Haskell;
tags/code/tags/ # contains Talon scripts which abstract over programming language features, e.g., comments.
```

I would be happy to submit these piece by piece—or in batches—to help the restructuring.

**Comment by rntz** [comment]
Yes (or at least, I'm interested), although a lot of bikeshedding is likely to ensue :). We might want to hash out the broad strokes of the reorganization in advance, either in this thread or in a community video meeting. At a glance your repository's organization looks good and thorough but more deeply nested than I tend to like.

If it's just moving files around, and we can get everyone on board with the organization scheme in advance, doing it all at once would be OK, although piecemeal would also be fine and probably easier to coordinate. If you want to change significantly more than file organization it would be good to have a list of the intended changes, and maybe do them PR-by-PR.

**Comment by wenkokke** [comment]
I think it tops out at 3 for `code/tags/XXX` or `apps/XXX/platforms`?

**Comment by fidgetingbits** [comment]
I'd love to see this change. Happy with the layout you already have personally, but not picky if others want to change things. 3 levels deep is fine by me.

**Comment by wenkokke** [comment]
Fixed in #897
